### PR TITLE
Imported tokens table sorting

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

Important tokens should be displayed together with SNS tokens that have a balance (between the important ck-tokens with a balance and ck-tokens without a balance). All tokens should be sorted alphabetically.

# Changes

- Replace `compareTokensWithBalanceFirst` with `compareTokensWithBalanceOrImportedFirst` comparator.
- Provide imported tokens id to the token table sorting function.

# Tests

- Unit tests were updated.

# Todos

- [ ] Add entry to changelog (if necessary).
Not yet.

### CODE REVIEW

File: frontend/src/lib/utils/tokens-table.utils.ts [hunk 1]
```diff
@@ -39,9 +46,15 @@ export const compareTokensAlphabetically = createAscendingComparator(
   ({ title }: UserToken) => title.toLowerCase()
 );
 
-export const compareTokensForTokensTable = mergeComparators([
-  compareTokensIcpFirst,
-  compareTokensWithBalanceFirst,
-  compareTokensByImportance,
-  compareTokensAlphabetically,
-]);
+// TODO: update unit tests
```
Comment (dskloetd) [inline frontend/src/lib/utils/tokens-table.utils.ts:49]: Should this be here?

File: frontend/src/tests/routes/app/tokens/page.spec.ts [hunk 2]
```diff
@@ -89,6 +92,20 @@ describe("Tokens route", () => {
     pending_utxos: [],
     required_confirmations: 0,
   });
+  const ckToken1Id = principal(100);
```
Comment (dskloetd) [inline frontend/src/tests/routes/app/tokens/page.spec.ts:95]: Why is it called `ck...`?
Would it be clearer to called it `imported...`?

File: frontend/src/tests/routes/app/tokens/page.spec.ts [hunk 3]
```diff
@@ -545,6 +565,86 @@ describe("Tokens route", () => {
...
+            ledgerCanisterId: ckToken1Id,
+            indexCanisterId: principal(111),
+          };
+          const ckToken2Data: ImportedTokenData = {
+            ledgerCanisterId: ckToken2Id,
+            indexCanisterId: undefined,
+          };
+
+          beforeEach(() => {
+            importedTokensStore.reset();
```
Comment (dskloetd) [inline frontend/src/tests/routes/app/tokens/page.spec.ts:579]: cleanups should always go in the top-level `beforeEach`.

File: frontend/src/tests/routes/app/tokens/page.spec.ts [hunk 4]
```diff
@@ -545,6 +565,86 @@ describe("Tokens route", () => {
...
+            const po = await renderPage();
+            const tokensPagePo = po.getTokensPagePo();
+            expect(await tokensPagePo.getTokenNames()).toEqual([
+              "Internet Computer",
+              // ck with balance
+              "ckBTC",
+              "ckUSDC",
+              // Imported tokens should be placed with the SNS tokens that have a non-zero balance
+              // and should be sorted alphabetically.
+              "ATOKEN2", // Imported
```
Comment (dskloetd) [inline frontend/src/tests/routes/app/tokens/page.spec.ts:638]: Is this one without balance?
Comment `// Imported without balance`?

File: frontend/src/lib/utils/tokens-table.utils.ts (hunk 1 above)
Comment (mstrasinskis) [reply frontend/src/lib/utils/tokens-table.utils.ts]: Good catch. Removed.

File: frontend/src/tests/routes/app/tokens/page.spec.ts (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/tests/routes/app/tokens/page.spec.ts:95]: done.

File: frontend/src/tests/routes/app/tokens/page.spec.ts (hunk 3 above)
Comment (mstrasinskis) [reply frontend/src/tests/routes/app/tokens/page.spec.ts:579]: Good point, done.

File: frontend/src/tests/routes/app/tokens/page.spec.ts (hunk 4 above)
Comment (mstrasinskis) [reply frontend/src/tests/routes/app/tokens/page.spec.ts:637]: The presence of a balance is not relevant for sorting the imported tokens.

Comment (dskloetd) [reply frontend/src/tests/routes/app/tokens/page.spec.ts:637]: Indeed, that's exactly the purpose of this PR.
And so you want to make sure that's tested.
It's already tested in the utils, but it's still nice to know that this token here is not moved to the bottom even though it doesn't have a balance.

**Review by dskloetd — APPROVED** [review]

File: frontend/src/tests/routes/app/tokens/page.spec.ts (hunk 4 above)
Comment (mstrasinskis) [reply frontend/src/tests/routes/app/tokens/page.spec.ts]: I see your point. Thanks. Updated.

**Pull request merged**